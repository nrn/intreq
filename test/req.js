var test = require('tape');
var intreq = require('../');

var input = require('./files/req.json');
var expected = require('./files/req_expected.json');

test(function (t) {
    t.plan(1);
    
    var r = intreq();
    var rows = [];
    r.on('data', function (row) { rows.push(row) });
    r.on('end', function () {
        t.deepEqual(rows, expected);
    });
    
    input.forEach(function (row) { r.write(row) });
    r.end();
});
