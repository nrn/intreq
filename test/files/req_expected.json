[
{"id":1,"source":"module.exports = function (n) { return n * 111 }\n\n","deps":{}}
,
{"id":2,"source":"var bar = require(1);\nmodule.exports = function (n) { return n * bar(1) };","deps":{"1":1}}
,
{"id":3,"source":"var foo = require(1);\n\nconsole.log(foo(5));","deps":{"1":2},"entry":true}
]
